compiler/rustc_typeck/src/astconv/mod.rs

```diff
@@ -2017,69 +2017,79 @@ impl<'o, 'tcx> dyn AstConv<'tcx> + 'o {
         self.normalize_ty(span, tcx.mk_projection(item_def_id, item_substs))
     }
 
-    pub fn prohibit_generics<'a, T: IntoIterator<Item = &'a hir::PathSegment<'a>>>(
+    pub fn prohibit_generics<'a, T: Iterator<Item = &'a hir::PathSegment<'a>> + Clone>(
```

Comment:
Isn't `T` always a slice `&[&hir::PathSegment<'_>]`? Can't we just take this?

Reply:
I tried doing that. It isn't, there are a couple of callers that pass in an Iterator after skipping a few things. I think at least one uses `filter_map`, which isn't amenable to a direct subslicing approach.